source/solvers/postprocessing_velocities.cc

```diff
@@ -12,26 +12,37 @@ AverageVelocities<dim, VectorType, DofsType>::calculate_average_velocities(
   const Parameters::PostProcessing &post_processing,
   const double &                    current_time,
   const double &                    time_step,
+  const bool &                      is_output_iteration,
   const DofsType &                  locally_owned_dofs,
   const MPI_Comm &                  mpi_communicator)
 {
   const double epsilon      = 1e-6;
   const double initial_time = post_processing.initial_time;
-  const double dt           = time_step;
+  dt                        = time_step;
 
   // When averaging velocities begins
-  if (current_time >= (initial_time - epsilon) && average_calculation == false)
+  if (current_time >= (initial_time - epsilon) && !average_calculation)
     {
       average_calculation = true;
       real_initial_time   = current_time;
 
       // Store the first dt value in case dt varies.
       dt_0 = dt;
 
-      // Reinitialisation of the vectors to get the right length
+      // Reinitialisation of the average velocity vectors to get the right
+      // length
       velocity_dt.reinit(locally_owned_dofs, mpi_communicator);
       sum_velocity_dt.reinit(locally_owned_dofs, mpi_communicator);
       average_velocities.reinit(locally_owned_dofs, mpi_communicator);
+
+      // Reinitialisation of reynolds stress vectors if calculation is enable
+      if (post_processing.calculate_reynolds_stress)
+        {
+          fluctuation.reinit(locally_owned_dofs, mpi_communicator);
+          reynolds_stress_dt.reinit(locally_owned_dofs, mpi_communicator);
+          sum_reynolds_stress_dt.reinit(locally_owned_dofs, mpi_communicator);
+          reynolds_stress.reinit(locally_owned_dofs, mpi_communicator);
```

Comment:
RIght now this means that you average velocity calculation won't work with dynamic mesh adaptation. You should at least document that because when we refine the mesh we need to "reinit the vectors and transfer the solution". It's not mandatory right now, but clearly this support should be added eventually.

Reply:
Do you think I should add a condition where if there's a dynamic mesh adaptation all vectors reinitializes at each iteration?